tests/test_presolve.py

```diff
@@ -190,8 +200,8 @@ def test_no_variable_constraints(self):
 
```

Comment:
I think an error should occur here rather than later in the presolve as we are trying to fix a variable at an impossible bound.

Reply:
Sorry, which line is this comment applying to?

Reply:
line 199, you need to expand the view. 
``` 
        cqm.add_constraint(i <= 1)
         cqm.fix_variable('i', 2)
```

Reply:
Ah, that would require a lot of logic in the CQM object itself and slow down construction significantly. The way we've designed the overall toolchain is optimizing for fast but permissive construction. Then, once the model is constructed, we can use presolve to do fast tests on feasibility. If we did eager error messages, we would end up checking the same things repeatedly which is redundant and slow.

Reply:
I was thinking of adding a check inside  cqm.fix_variable() method, and simply check the upper and lower bound of variable and giving error if one try to fix a variable beyond it's range. Is that still expensive? I know this is a feature request and not related to this PR.